File: LeetCode_2044/Main.java
public class Main{
    public static void main(String[] args) {
        
    }
}

class Solution {
    public int countMaxOrSubsets(int[] nums) {
        int maxOr = 0, count = 0;
        int n = nums.length;
        int total = 1 << n;

        for (int mask = 0; mask < total; mask++) {
            int cur = 0;
            for (int i = 0; i < n; i++) {
                if ((mask & (1 << i)) != 0) {
                    cur |= nums[i];
                }
            }
            if (cur > maxOr) {
                maxOr = cur;
                count = 1;
            } else if (cur == maxOr) {
                count++;
            }
        }
        return count;
    }
}

File: LeetCode_2044/Main.py
class Solution(object):
    def countMaxOrSubsets(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        total = 1 << n
        max_or = 0
        count = 0

        for mask in range(total):
            cur = 0
            for i in range(n):
                if (mask & (1 << i)) != 0:
                    cur |= nums[i]
            if cur > max_or:
                max_or = cur
                count = 1
            elif cur == max_or:
                count += 1

        return count


